### PR TITLE
Updated README.md to relect current behavior and have instructions

### PR DESCRIPTION
**Pull request by android-usb**
for adding support for Stereo input samples (though not the
implementation).

Issue #801

### CODE REVIEW

**Comment by android-usb** [comment]
Issue #913

**Comment by android-usb** [comment]
Gerry, could you see what's going wrong with travis here?

File: samples/drumthumper/README.md [hunk 1]
```diff
@@ -24,12 +24,19 @@ Secondarily, **DrumThumper** demonstrates:
 * A mechanism for parsing/loading one type (WAV) of audio data.
 * How to control the relative levels (gain) of audio sources mixed into an output stream.
 * How to locate a mono data source in a stereo output stream.
+* How to use the Oboe resampler to resample source audio to the device playback rate, and therefore not incur this overhead at playback time.
 
 To keep things simple, **DrumThumper** specifically does not:
-* Does not provide support audio samples in other than 16-bit, 44.1K, mono PCM Samples. It does not support Stereo, different samples rates or different PCM formats.
+* Does not provide support audio samples in other than 16-bit, mono PCM Samples. It does not support Stereo or different PCM formats.
```
Comment (philburk) [inline samples/drumthumper/README.md:30]: remove "provide"

File: samples/drumthumper/README.md [hunk 2]
```diff
@@ -24,12 +24,19 @@ Secondarily, **DrumThumper** demonstrates:
...
 * Does not provide support for non-WAV audio data (such as AIFF).
 * Does not provide support for compressed audio data.
 
+**DrumThumper** does now support different sample rates for the source samples.
```
Comment (philburk) [inline samples/drumthumper/README.md:34]: I read this as "does not". Maybe say "now support"

File: samples/drumthumper/README.md [hunk 3]
```diff
@@ -24,12 +24,19 @@ Secondarily, **DrumThumper** demonstrates:
 * A mechanism for parsing/loading one type (WAV) of audio data.
 * How to control the relative levels (gain) of audio sources mixed into an output stream.
 * How to locate a mono data source in a stereo output stream.
+* How to use the Oboe resampler to resample source audio to the device playback rate, and therefore not incur this overhead at playback time.
 
 To keep things simple, **DrumThumper** specifically does not:
```
Comment (philburk) [inline samples/drumthumper/README.md:29]: Double "does not"

File: samples/drumthumper/README.md [hunk 4]
```diff
@@ -24,12 +24,19 @@ Secondarily, **DrumThumper** demonstrates:
...
-* Does not provide support audio samples in other than 16-bit, 44.1K, mono PCM Samples. It does not support Stereo, different samples rates or different PCM formats.
+* Does not provide support audio samples in other than 16-bit, mono PCM Samples. It does not support Stereo or different PCM formats.
 * Does not provide support for non-WAV audio data (such as AIFF).
 * Does not provide support for compressed audio data.
 
+**DrumThumper** does now support different sample rates for the source samples.
+
+If an one wanted to extend **DrumThumper** to support Stereo samples, one would need to:
```
Comment (philburk) [inline samples/drumthumper/README.md:36]: Double "would need to"

File: samples/iolib/src/main/cpp/player/SampleBuffer.cpp [hunk 5]
```diff
@@ -64,7 +64,7 @@ void resampleData(const ResampleBlock& input, ResampleBlock* output) {
     // so add a few more frames for padding
     numOutFrames += 8;
 
-    const int channelCount = 1;    // 1 for mono, 2 for stereo
+    const int channelCount = numChannels;    // 1 for mono, 2 for stereo
```
Comment (philburk) [inline samples/iolib/src/main/cpp/player/SampleBuffer.cpp:67]: Can we just pass numChannels to make()?

**Review by philburk — CHANGES_REQUESTED** [review]

**Comment by ggfan** [comment]
@android-usb it says running too long and timeout: last time it happens and then later it cured itself.  after you get review feedback addressed, you will automatically trigger a re-run, we could check it then?

File: samples/drumthumper/README.md (hunk 2 above)
Comment (android-usb) [reply samples/drumthumper/README.md:34]: "now supports"

File: samples/drumthumper/README.md (hunk 1 above)
Comment (android-usb) [reply samples/drumthumper/README.md:30]: Done.

File: samples/drumthumper/README.md (hunk 4 above)
Comment (android-usb) [reply samples/drumthumper/README.md:36]: Done.

File: samples/drumthumper/README.md (hunk 3 above)
Comment (android-usb) [reply samples/drumthumper/README.md:29]: Done.

File: samples/iolib/src/main/cpp/player/SampleBuffer.cpp (hunk 5 above)
Comment (android-usb) [reply samples/iolib/src/main/cpp/player/SampleBuffer.cpp:67]: Yes.

File: samples/drumthumper/README.md [hunk 6]
```diff
@@ -24,11 +24,18 @@ Secondarily, **DrumThumper** demonstrates:
...
-* Does not provide support audio samples in other than 16-bit, 44.1K, mono PCM Samples. It does not support Stereo, different samples rates or different PCM formats.
-* Does not provide support for non-WAV audio data (such as AIFF).
-* Does not provide support for compressed audio data.
+* Does not support audio samples in other than 16-bit, mono PCM Samples. It does not support Stereo or different PCM formats.
+* Does not support for non-WAV audio data (such as AIFF).
+* Does not support for compressed audio data.
+
+**DrumThumper** now supports different sample rates for the source samples.
+
+If an one wanted to extend **DrumThumper** to support Stereo samples:
```
Comment (philburk) [inline samples/drumthumper/README.md:36]: c/an/any/

**Review by philburk — APPROVED** [review]

File: samples/drumthumper/README.md [hunk 7]
```diff
@@ -24,11 +24,18 @@ Secondarily, **DrumThumper** demonstrates:
...
 * How to control the relative levels (gain) of audio sources mixed into an output stream.
 * How to locate a mono data source in a stereo output stream.
+* How to use the Oboe resampler to resample source audio to the device playback rate, and therefore not incur this overhead at playback time.
 
 To keep things simple, **DrumThumper** specifically does not:
-* Does not provide support audio samples in other than 16-bit, 44.1K, mono PCM Samples. It does not support Stereo, different samples rates or different PCM formats.
-* Does not provide support for non-WAV audio data (such as AIFF).
-* Does not provide support for compressed audio data.
+* Does not support audio samples in other than 16-bit, mono PCM Samples. It does not support Stereo or different PCM formats.
+* Does not support for non-WAV audio data (such as AIFF).
```
Comment (philburk) [inline samples/drumthumper/README.md:31]: Remove "for".

**Review by philburk — CHANGES_REQUESTED** [review]

**Review by philburk — APPROVED** [review]

**Pull request merged**